src/queries/release.tsx

```diff
@@ -20,24 +28,42 @@ function useReleasesQuery(
   const finalParams: RepositoryQueryParams = mapRepositoryToQueryParams(
     params.repository ?? undefined
   )
+  const { fromVersion, toVersion } = params
+  const hasFromVersion = !!fromVersion
+  const hasToVersion = !!toVersion
 
   return useQuery<ReleasesQueryResults, Error>(
     [QUERY_KEY, finalParams],
     async () => {
+      const releases: Array<Release> = []
       let paginationCount = 0
-      return octokit.paginate(
+
+      for await (const response of octokit.paginate.iterator(
         'GET /repos/{owner}/{repo}/releases',
-        { ...finalParams, per_page: 100 },
-        (response, done) => {
-          paginationCount++
+        { ...finalParams, per_page: 100 }
+      )) {
+        paginationCount++
+        releases.push(...response.data.filter(isStableRelease))
 
-          if (paginationCount === MAX_AUTO_PAGINATION_ALLOWED) {
-            done()
-          }
+        const isMaxAutoPaginationReached =
+          paginationCount === MAX_AUTO_PAGINATION
+        const lastReleaseFetched = response.data[response.data.length - 1]
+        const isFromReleaseFetched =
+          !hasFromVersion ||
+          semver.gte(fromVersion, lastReleaseFetched.tag_name)
+        const isToReleaseFetched =
+          !hasToVersion || semver.gte(toVersion, lastReleaseFetched.tag_name)
```

Comment:
This is basically where the magic happens to check if both from and to versions have been fetched already.